File: server/lib/comcent/types/deepgram_result.ex
defmodule Comcent.Types.DeepgramResult do
  @moduledoc """
  Represents the structure of a Deepgram transcription result.
  """

  defstruct [
    :results
  ]

  @type t :: %__MODULE__{
          results: %{
            channels: [
              %{
                alternatives: [
                  %{
                    paragraphs: %{
                      paragraphs: [
                        %{
                          sentences: [
                            %{
                              text: String.t(),
                              start: number()
                            }
                          ]
                        }
                      ]
                    }
                  }
                ]
              }
            ]
          }
        }
end

File: packages/web-app/src/lib/server/types/DeepgramResult.ts
// data.results.channels[0].alternatives[0].paragraphs

export type DeepgramResult = {
  results: {
    channels: [
      {
        alternatives: [
          {
            paragraphs: {
              paragraphs: [
                {
                  sentences: {
                    text: string;
                    start: number;
                  }[];
                },
              ];
            };
          },
        ];
      },
    ];
  };
};
